QuantConnect.TradeStationBrokerage/TradeStationBrokerage.cs

```diff
@@ -680,6 +663,98 @@ await foreach (var json in _tradeStationApiClient.StreamOrders())
         return _autoResetEvent.WaitOne(TimeSpan.FromSeconds(10), _cancellationTokenSource.Token);
     }
 
+    /// <summary>
+    /// Handles incoming TradeStation messages in JSON format.
+    /// </summary>
+    /// <param name="json">The JSON string containing the TradeStation message.</param>
+    private void HandleTradeStationMessage(string json)
+    {
+        var jObj = JObject.Parse(json);
+        if (IsConnected && jObj["AccountID"] != null)
```

Comment:
- `IsConnected` required when handling a message? doesn't seem right
- Pending TODO test live trading internet disconnections & reconnection, long live deployments, over the weekend

Reply:
> - IsConnected required when handling a message? doesn't seem right

Yes, we're using it correctly. We wait for the order snapshot to complete, during which `IsConnected` is **false**.
Once the snapshot is done, `IsConnected` turns **true**, and we start getting real-time updates.
So, handling messages when `IsConnected` is **true** ensures we're working with live data.

Reply:
> Pending TODO test live trading internet disconnections & reconnection, long live deployments, over the weekend
- [x] internet disconnections & reconnection
- [ ] long live deployments, over the weekend